models/issues/issue.go

```diff
@@ -448,6 +452,59 @@ func (issue *Issue) IsPoster(uid int64) bool {
 	return issue.OriginalAuthorID == 0 && issue.PosterID == uid
 }
 
+func (issues IssueList) FilterValidByDoer(ctx context.Context, doer *user_model.User) (IssueList, error) {
+	repos, err := issues.LoadRepositories(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := repo_model.RepositoryList(repos).LoadOwners(ctx); err != nil {
+		return nil, err
+	}
+
+	issueList := issues[:0]
+	for _, issue := range issues {
+		if isIssueVisibleToDoer, err := issue.isIssueVisibleToDoer(ctx, doer); err != nil {
+			return nil, err
+		} else if isIssueVisibleToDoer {
+			issueList = append(issueList, issue)
+		}
+	}
+	return issueList, nil
+}
+
+// isIssueVisibleToDoer returns whether doer can view the issue
+func (issue *Issue) isIssueVisibleToDoer(ctx context.Context, doer *user_model.User) (bool, error) {
+	if perm, err := access_model.GetUserRepoPermission(ctx, issue.Repo, doer); err != nil {
```

Comment:
The permission check is for a repository not only for one issue. It's a performance problem. I think it has to be changed.

Reply:
Done.